### PR TITLE
Basic queue data structure to handle FIFO scheduling of plans.

### PR DESCRIPTION
**Pull request by thanset**
Signed-off-by: Tim Hansen <timothy.hansen@one.verizon.com>

### CODE REVIEW

**Comment by aaronjwood** [comment]
Do you think we could add a generic version of the queue to the `structures` dir in the SDK and turn it into a plan manager in this repo?

**Comment by thanset** [comment]
@aaronjwood sure! not a problem. That's probably a better idea.

**Comment by aaronjwood** [comment]
Cool, just figured other people could use it that way!

File: task/plan/planManager.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+func (p *PlanManager) Push(planType PlanType) {
+	p.current = append(p.current, planType)
+}
+
+func (p *PlanManager) Pop() PlanType {
+	curr := Idle
+	if len(p.current) > 0 {
+		curr = p.current[0]
+		p.current = p.current[1:len(p.current)]
```
Comment (aaronjwood) [inline task/plan/planManager.go]: This is such a minor thing but we could do `p.current = p.current[1:]` to make it easier to read. If you want to keep it explicit like it is now that's fine too.

**Comment by thanset** [comment]
@aaronjwood I will add this into the SDK once I finish implementing the plan structure in sprint.

**Pull request merged**